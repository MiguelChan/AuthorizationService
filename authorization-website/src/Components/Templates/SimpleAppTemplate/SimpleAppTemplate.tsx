import { AppBar, Container, Paper, useTheme } from '@mui/material';
import React from 'react';

export interface SimpleAppTemplateProps {
  applicationToolbar?: React.ReactNode;
  applicationContent?: React.ReactNode;
}

export const SimpleAppTemplate: React.FunctionComponent<SimpleAppTemplateProps> = ({
  applicationToolbar,
  applicationContent,
}) => {

  const currentTheme = useTheme();

  const getApplicationToolbar = (): React.ReactNode => {
    if (applicationToolbar === null || applicationToolbar === undefined) {
      return (
        <>ApplicationToolbarPosition</>
      );
    }

    return applicationToolbar!;
  };

  const getApplicationContent = (): React.ReactNode => {
    if (applicationContent === null || applicationContent === undefined) {
      return (
        <>ApplicatonContentPosition</>
      );
    }
    return applicationContent!;
  }

  return (
    <>
      <AppBar position='static'>
        {getApplicationToolbar()}
      </AppBar>
      <Container maxWidth='xl' sx={{marginTop: currentTheme.spacing(2)}}>
        <Paper sx={{padding: currentTheme.spacing(1)}}>
          {getApplicationContent()}
        </Paper>
      </Container>
    </>
  );
};