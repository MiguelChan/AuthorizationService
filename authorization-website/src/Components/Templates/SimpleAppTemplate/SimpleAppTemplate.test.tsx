import React from 'react';
import {
  render,
  RenderResult,
} from '@testing-library/react';
import { SimpleAppTemplate } from './SimpleAppTemplate';

describe('SimpleAppTemplate', () => {
  it('Should display the Template', () => {
    const renderResult = setupComponent();  
    expect(renderResult.container).toMatchSnapshot();
  });

  const setupComponent = (): RenderResult => {
    return render(<SimpleAppTemplate />);
  }

});