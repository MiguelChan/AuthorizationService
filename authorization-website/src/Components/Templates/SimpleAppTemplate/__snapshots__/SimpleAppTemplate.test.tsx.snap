// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleAppTemplate Should display the Template 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    ApplicationToolbarPosition
  </header>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl css-qhkpqj-MuiContainer-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-3ze8r2-MuiPaper-root"
    >
      ApplicatonContentPosition
    </div>
  </div>
</div>
`;
